### PR TITLE
add check for message fields when determining message type

### PR DESCRIPTION
**Pull request by dirk-thomas**
This addresses #376 - or at least the feasible part of it.

@esteve @tfoote @wjwwood Please provide feedback.

### CODE REVIEW

File: tools/rostopic/src/rostopic/__init__.py [hunk 1]
```diff
@@ -307,6 +307,27 @@ def _get_topic_type(topic):
...
+            msg_class = roslib.message.get_message_class(t_type)
+            if not msg_class:
+                # if any class is not fetchable skip ignoring any message types
+                break
+            msg = msg_class()
+            pattern = topic[len(t):].rstrip('/')
+            if pattern == '':
+                break
+            try:
+                eval('msg'+'.'.join(pattern.split('/')))
```
Comment (tfoote) [inline tools/rostopic/src/rostopic/__init__.py]: Is this sanitized sufficiently at this point to be safe?

Comment (dirk-thomas) [inline tools/rostopic/src/rostopic/__init__.py]: The same logic is already used in line 287 but I do agree some kind of sanity checking would be nice.

**Comment by wjwwood** [comment]
+1 code review

**Comment by jihoonl** [comment]
This addresses the issue https://github.com/RobotWebTools/rosbridge_suite/issues/85

**Comment by esteve** [comment]
+1

**Comment by dirk-thomas** [comment]
In my recent commit I replaced both uses of `eval()` with a loop of `getattr()` calls. That is indeed safer - I was able to craft some really interesting `rostopic ...` command lines :wink:

**Comment by tfoote** [comment]
+1

**Pull request merged**